wiki/development/coding-guidelines.md

```diff
@@ -8,27 +8,185 @@ order: 1
 ---
 
 
-## 1. Indentation
+## 1. Naming Conventions
 
-4 spaces for indentation.
+### 1.1. Variable Names
+#### 1.1.1. Global Variable naming
+All global variables must start with the ACE prefix followed by the component, separated by underscores. Global variables may not contain the fnc_ prefix if the value is not callable code.
 
-```c++
-class Something: Or {
-....class Other {
-........foo = "bar";
-....};
+Example: `ACE_Component_myVariableName`
+
+_For ACE this is done automatically through the usage of the GVAR macro family._
+
+#### 1.1.2. Function naming
+All functions shall use ACE and the component name as a prefix, as well as the fnc_ prefix behind the component name.
+
+Example: `PREFIX_COMPONENT_fnc_functionName`
+
+_For ACE this is done automatically through the usage of the PREP macro_
+
+### 1.2. Files & Config
+
+#### 1.2.1. SQF File extension
+Files containing sqf scripts will always have a file name extension of ".sqf".
```

Comment:
Should be decided overall if "will" or "shall" or whatever should be used. I vote "shall" or "should"